File: dms-web/src/main/java/com/dms/entity/Menu.java
package com.dms.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.dms.core.domain.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 菜单实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("sys_menu")
public class Menu extends BaseEntity {
    private String menuCode; // 菜单编码（唯一标识）
    private String menuName; // 菜单名称
    private String menuType; // 菜单类型：menu-菜单 button-按钮
    private String path; // 路由路径
    private String component; // 组件路径
    private String icon; // 图标
    private Long parentId; // 父菜单ID，0表示顶级菜单
    private Integer sortOrder; // 排序
    private Integer status; // 状态：0-禁用 1-启用
    private String description; // 描述
}

File: frontend/src/api/menu.ts
import request from '@/utils/request'

export interface Menu {
  id?: number
  menuCode: string
  menuName: string
  menuType?: string
  path?: string
  component?: string
  icon?: string
  parentId?: number
  sortOrder?: number
  status?: number
  description?: string
  children?: Menu[]
}

// 获取菜单树
export function getMenuTree() {
  return request.get<Menu[]>('/menu/tree')
}

// 根据ID获取菜单
export function getMenuById(id: number) {
  return request.get<Menu>(`/menu/${id}`)
}

// 创建菜单
export function createMenu(menu: Menu) {
  return request.post('/menu', menu)
}

// 更新菜单
export function updateMenu(menu: Menu) {
  return request.put('/menu', menu)
}

// 删除菜单
export function deleteMenu(id: number) {
  return request.delete(`/menu/${id}`)
}

// 获取当前用户的菜单列表
export function getCurrentUserMenus() {
  return request.get<Menu[]>('/user/my/menus')
}

